### PR TITLE
Try both Clevis and passphrase in /etc/fstab setup

### LINKED ISSUE #3349: Pools with NBDE can never be unlocked with a passphrase during boot
A encrypted pool that should be unlocked during boot can be bound to a Tang server so that the boot can be unattended. However, if the Tang server is not reachable during boot, the stratis-fstab-setup program will just fail and not allow entering a passphrase at all.

This is different from LUKS. With LUKS, you get the opportunity to enter a passphrase even when a device uses NBDE as well.

### PR DESCRIPTION
**Pull request by jbaublitz**
Closes #3349

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mvollmer I tested this patch and it resolves the linked issue. The one question I had was mainly about whether you had this problem at boot time too. I think we may take a similar approach but I'm not sure.

**Comment by jbaublitz** [comment]
@mulkieran I'm putting this in review after testing and bug fixes.

**Review by mvollmer — COMMENTED** [review]
I think this works, but the code was a bit hard to follow.

**Comment by jbaublitz** [comment]
@mvollmer I've re-requested your review. I think I've resolved the issues you mentioned.

**Review by mulkieran — APPROVED** [review]

**Comment by jbaublitz** [comment]
@mvollmer After investigating a little bit more, I can confirm that the same problem affects the initrd scripts, but it's a little bit harder to solve there largely because we have intentionally separated the Clevis unlock method and the passphrase unlock method into separate dracut modules. This is largely because Anaconda does not support Clevis but does support the keyring unlock method for cryptsetup. These are distributed in separate packages as a result so Anaconda could theoretically support one without the other. I'll have to look into whether there's a way to set a flag based on whether executables are present or not in a dracut module. If we did that, we could potentially merge both dracut modules into a single, dynamic one that would also satisfy the requirement for Anaconda.

**Comment by mulkieran** [comment]
@mvollmer We're ready to merge as soon as you're satisfied w/ the changes.

**Review by mvollmer — APPROVED** [review]
Thanks!
